pkg/util/http.go

```diff
@@ -0,0 +1,44 @@
+/*
+Copyright 2014 Google Inc. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package util
+
+import (
+	"io"
+	"net/url"
+	"strings"
+)
+
+// IsProbableEOF returns true if the given error resembles a connection termination
+// scenario that would justify assuming that the watch is empty. The watch stream
```

Comment:
This comment is out of date a bit - split the comment into two parts - move the watch part back into request.go (right before we call isProbableEOF), and here explain that these are errors that the Go http stack returns back to us which are general connection closure errors (strongly correlated) and that callers that need to differentiate probable errors in connection behavior between normal "this is disconnected" should use the method.